src/main/java/org/jetbrains/research/refactorinsight/services/MiningService.java

```diff
@@ -187,28 +197,75 @@ public void mineAndWait(GitRepository repository) {
   public void mineAtCommit(VcsCommitMetadata commit, Project project, GitWindow info) {
     ProgressManager.getInstance()
         .run(new Task.Backgroundable(project, String.format(
-            RefactoringsBundle.message("mining.at"), commit.getId().asString())) {
+            RefactorInsightBundle.message("mining.at"), commit.getId().asString())) {
 
           @Override
-          public void onCancel() {
-            super.onCancel();
-          }
-
           public void onFinished() {
-            super.onFinished();
             if (contains(commit.getId().asString())) {
-              System.out.println(RefactoringsBundle.message("finished"));
               ApplicationManager.getApplication()
                   .invokeLater(() -> info.refresh(commit.getId().asString()));
             }
           }
 
+          @Override
           public void run(@NotNull ProgressIndicator progressIndicator) {
-              CommitMiner.mineAtCommitTimeout(commit, innerState.refactoringsMap.map, project);
+            try {
+              runWithCheckCanceled(() -> {
+                    CommitMiner.mineAtCommitTimeout(commit, innerState.refactoringsMap.map, project);
+                    return null;
+                  },
+                  progressIndicator);
+            } catch (Exception e) {
+              e.printStackTrace();
```

Comment:
Probably you should use `Logger.error("Message", e)`, not just printStackTrace.

Reply:
Yeah, we definitely should use Logger in the project! I will take care of it.